public/index.html

```diff
@@ -0,0 +1,82 @@
+<!-- public/index.html -->
+<!DOCTYPE html>
+<html lang="en">
+<head>
+<meta charset="UTF-8">
+<title>Chat App</title>
+<style>
+  body {
+    background-color: black;
+    color: white;
+    font-family: Arial, sans-serif;
+  }
+  #chat {
+    max-width: 600px;
+    margin: 20px auto;
+    padding: 10px;
+    border: 1px solid #333;
+  }
+  #messages {
+    height: 300px;
+    overflow-y: auto;
+    margin-bottom: 10px;
+    padding: 5px;
+    background: #111;
+  }
+  #messageInput {
+    width: calc(100% - 60px);
+    margin-right: 10px;
+    background-color: dimgray;
+    color:white;
+  }
+  button {
+    width: 50px;
+    background-color: dimgray;
+    color:white;
+  }
+</style>
+</head>
+<body>
+<h1> NodeJS Cortext sample app </h1>
+<div id="chat">
+  <div id="messages"></div>
+  <input type="text" id="messageInput">
+  <button onclick="sendMessage()">Send</button>
+</div>
+
+<script>
+let conversationHistory = [];
+
+function sendMessage() {
```

Comment:
**Issue**: The sendMessage function directly manipulates the DOM for every message sent, which can lead to performance issues on larger scales. Each call to innerHTML causes the browser to re-parse the HTML, which can be inefficient. <br> **Fix**: Use document.createElement and appendChild instead of innerHTML to update the DOM. This approach avoids re-parsing the entire inner HTML of the messages div every time a new message is added. <br> **Code Suggestion**: 
 ```
 -  messagesDiv.innerHTML += '<div><b>User</b>: ${userMessage}</div>';
 +  const messageElement = document.createElement('div');
 +  messageElement.innerHTML = '<b>User</b>: ${userMessage}';
 +  messagesDiv.appendChild(messageElement);
 ```